src/arrayMethodSort.js

```diff
@@ -4,8 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
+      comparator = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+    } else {
+      comparator = compareFunction;
+    }
+
+    for (let i = 0; i < length - 1; i++) {
+      for (let j = 0; j < length - 1 - i; j++) {
+        if (comparator(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
```

Comment:
You can use destructuring to swap the elements. It will make your code more readable and concise. Like so: `[this[j], this[j + 1]] = [this[j + 1], this[j]];`.